api/src/main/java/org/eclipse/microprofile/context/ThreadContext.java

```diff
@@ -511,13 +518,13 @@ public void foo() {
      * <p>Returns a new <code>CompletionStage</code> that is completed by the completion of the
      * specified stage.</p>
      *
-     * <p>The new completion stage is backed by a ThreadContext rather than a ManagedExecutor.
-     * Therefore, the new stage and all dependent stages created from it, and so forth,
-     * have no default asynchronous execution facility and must raise {@link java.lang.UnsupportedOperationException}
-     * for all <code>*Async</code> methods that do not specify an executor. For example,
-     * {@link java.util.concurrent.CompletionStage#thenRunAsync(Runnable) thenRunAsync(Runnable)}.</p>
-     *
-     * <p>When dependent stages are created from the new completion stage, thread context is captured
+     * <p>The new completion stage will use the same default executor as this ThreadContext,
+     * which can be a ManagedExecutor if this ThreadContext was obtained by {@link ManagedExecutor#getThreadContext()}
+     * or the default executor service as set by
+     * {@link ContextManager.Builder#withDefaultExecutorService(ExecutorService)},
+     * or otherwise have no default executor.</p>
+     * 
+     * <p>If this thread context has no default executor, the new stage and all dependent stages created from it, and so forth,
```

Comment:
It looks like some of the JavaDoc after this point was accidentally cut out.

Reply:
Well spotted.